### PR TITLE
Add support for use Agent Config File

### PR DESCRIPTION
**Pull request by 0lvin**
Add ability to use "dynamic" connection, when before call if ClientConfig has some value in AgentFile filed - will try to load agent config from fs.

### CODE REVIEW

**Review by EarthmanT — APPROVED** [review]

File: cloudify/client.go [hunk 1]
```diff
@@ -23,26 +23,55 @@ import (
...
+	Password  string
+	Tenant    string
+	Debug     bool
+	AgentFile string
+}
+
 //Client - struct with connection settings for connect to manager
 type Client struct {
-	restCl rest.ConnectionOperationsInterface
+	ClientConfig
```
Comment (mabuaisha) [inline cloudify/client.go:38]: @0lvin Do not you think that ClientConfig should be something like this 
```
ClientConfigObj  ClientConfig
```
I mean thats the way how we define variable inside struct on go

File: cloudify/tests/asserts.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+Package tests - fake classes for api testing.
+*/
+package tests
+
+import (
+	"testing"
+)
+
+// AssertEqual - check values
+func AssertEqual(t *testing.T, value, needed interface{}, format string, v ...interface{}) {
```
Comment (mabuaisha) [inline cloudify/tests/asserts.go:27]: @0lvin  There is a package available for assertion and we do not need to add a new one , I used it before check it https://godoc.org/github.com/stretchr/testify/assert

**Review by mabuaisha — COMMENTED** [review]
@0lvin I just leave two tiny comments

File: cloudify/tests/asserts.go (hunk 2 above)
Comment (0lvin) [reply cloudify/tests/asserts.go:27]: For use it need to put it to vendor. And we can go without additional package if we use only one really simple func from it.

File: cloudify/client.go (hunk 1 above)
Comment (0lvin) [reply cloudify/client.go:38]: I like to inherit from ClientConfig(not include as field).

File: cloudify/tests/asserts.go (hunk 2 above)
Comment (0lvin) [reply cloudify/tests/asserts.go:27]: And yes of course, when we will use mock logic from testify - we add testify as git submodule to vendor. Without such we broke "make test" on https://github.com/cloudify-incubator/cloudify-kubernetes-provider/blob/master/Makefile#L149

Comment (mabuaisha) [reply cloudify/tests/asserts.go:27]: Thanks For clarification

**Pull request merged**